### ISSUE TITLE
Inliner + printf = bugs

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#395]
```
import 'regent'
local __demand(__inline)
task foo()
  return 1.0
end
task main()
  -- BUG 1: This doesn't get printed
  regentlib.c.printf('%lf\n', foo())
end
-- BUG 2: Uncommenting this causes a compile-time crash:
-- .../src/regent/pretty.t:677: attempt to call method 'is' (a nil value)
-- main:printpretty()
regentlib.start(main)
```

**Comment by magnatelee** [comment]
This should be fix by commit e83246a1. @manopapad can close this issue if he confirms this is fixed.

**Comment by manopapad** [comment]
Issue is fixed.

**Comment by manopapad** [comment]
There is a different issue after this commit, which shows up when I try to use my HDF helper library. For example, when I try running the reproducer from issue #381, I get the following error:

```
sapling:~/minus_one_is_bad> ./test.sh
/home/manolis/legion/language/src/regent/symbol_table.t:59: Errors reported during typechecking.
test.rg:21: name '$__dump_ret' is undefined or nil
  dump(tiles, 'out_dir', grid, grid_copy, p_grid, p_grid_copy)
      ^

stack traceback:
	[C]: in function 'error'
	src/terralib.lua:388: in function 'finishandabortiferrors'
	/home/manolis/legion/language/src/common/report.t:52: in function 'error'
	/home/manolis/legion/language/src/regent/symbol_table.t:59: in function 'lookup'
	/home/manolis/legion/language/src/regent/type_check.t:422: in function 'expr'
	/home/manolis/legion/language/src/regent/type_check.t:3576: in function 'fn'
	src/asdl.lua:18: in function 'map'
	/home/manolis/legion/language/src/regent/type_check.t:3162: in function 'block'
	/home/manolis/legion/language/src/regent/type_check.t:3769: in function 'top_task'
	/home/manolis/legion/language/src/regent/type_check.t:3836: in function </home/manolis/legion/language/src/regent/type_check.t:3834>
	/home/manolis/legion/language/src/regent/passes.t:56: in function </home/manolis/legion/language/src/regent/passes.t:48>
	test.rg:13: in main chunk
```

**Comment by magnatelee** [comment]
I believe this is fixed in commit 058486e9.

**Comment by manopapad** [comment]
It works now, thanks!
